lib/takeaway.rb

```diff
@@ -0,0 +1,39 @@
+require_relative './checkout.rb'
+
+class Takeaway
+  attr_reader :menu, :basket
+
+  def initialize
+    @menu = {
+      'bibimbap'=>10.50,
+      'bulgogi'=>10,
+      'naengmyeon'=>8
+    }
+    @basket = []
+  end
+
+  def add_to_basket(item, quantity)
+    self.menu.map do |food, price|
+      if item == food
+        @basket.push("#{item} x #{quantity.to_s}: £")
+        @basket.push((price * quantity))
+      end
+    end
```

Comment:
This block of code:
1. finds the price of the given item
2. pushes two strings into the `@basket` instance variable.


For 1:
Hashes have a nice way of doing lookup using the square bracket syntax: `self.menu[item]` will give you back the corresponding price of the item

Reply:
Currently the `@basket` is something that stores strings which will be printed out to the user, which means that we're coupling the storing of items to how we want to present them to the user later on. The basket is currently doing two jobs:
- storing the items we want
- caring about how those items will be later printed to the user

Rather than storing strings, we could store dishes that the user ordered. Then we could have a method which translated this list of dishes into the format that we wanted to show the user. This would mean that the two concerns are separated.